test/bigtable/row.js

```diff
@@ -0,0 +1,785 @@
+    var timestamp = Date.now();
+
+    var families = [{
+      name: 'test-family',
+      columns: [{
+        qualifier: 'test-column',
+        cells: [{
+          value: 'test-value',
+          timestampMicros: timestamp,
+          labels: []
+        }]
+      }]
+    }];
+
+    var formattedRowData = {
+      'test-family': {
+        'test-column': [{
+          value: 'test-value',
+          timestamp: timestamp,
+          labels: []
+        }]
+      }
+    };
+
+    it('should format the families into a user-friendly format', function() {
+      var formatted = Row.formatFamilies_(families);
+
+      assert.deepEqual(formatted, formattedRowData);
+      assert.strictEqual(FakeMutation.convertFromBytes.callCount, 2);
+      assert(FakeMutation.convertFromBytes.getCall(0).args[0], 'test-column');
+      assert(FakeMutation.convertFromBytes.getCall(1).args[0], 'test-value');
+    });
+  });
+
+  describe('create', function() {
+    it('should provide the proper request options', function(done) {
+      row.parent.mutate = function(entry) {
+        assert.strictEqual(entry.key, row.id);
+        assert.deepEqual(entry.data, {});
+        assert.strictEqual(entry.method, Mutation.methods.INSERT);
+        done();
+      };
+
+      row.create(assert.ifError);
+    });
+
+    it('should accept data to populate the row', function(done) {
+      var data = {
+        a: 'a',
+        b: 'b'
+      };
+
+      row.parent.mutate = function(entry) {
+        assert.strictEqual(entry.data, data);
+        done();
+      };
+
+      row.create(data, assert.ifError);
+    });
+
+    it('should return an error to the callback', function(done) {
+      var err = new Error('err');
+      var response = {};
+
+      row.parent.mutate = function(entry, callback) {
+        callback(err, response);
+      };
+
+      row.create(function(err_, row, apiResponse) {
+        assert.strictEqual(err, err_);
+        assert.strictEqual(row, null);
+        assert.strictEqual(response, apiResponse);
+        done();
+      });
+    });
+
+    it('should return the Row instance', function(done) {
+      var response = {};
+
+      row.parent.mutate = function(entry, callback) {
+        callback(null, response);
+      };
+
+      row.create(function(err, row_, apiResponse) {
+        assert.ifError(err);
+        assert.strictEqual(row, row_);
+        assert.strictEqual(response, apiResponse);
+        done();
+      });
+    });
+  });
+
+  describe('createRules', function() {
+    var rules = [{
+      column: 'a:b',
+      append: 'c',
+      increment: 1
+    }];
+
+    it('should read/modify/write rules', function(done) {
+      row.request = function(grpcOpts, reqOpts) {
+        assert.deepEqual(grpcOpts, {
+          service: 'BigtableService',
+          method: 'readModifyWriteRow'
+        });
+
+        assert.strictEqual(reqOpts.tableName, TABLE.id);
+        assert.strictEqual(reqOpts.rowKey, ROW_ID);
```

Comment:
I think we need to check that this is the result of calling `Mutation.convertToBytes(this.id)`.